images/cfw/opt/x1plusd.py

```diff
@@ -1,116 +1,282 @@
+            "buildTimestamp", 0
+        ):
+            self.ota_available = True
+
+        # Return result of our check
+        return json.dumps({"ota": ota_response})
+
+
 class SettingsService:
-    DEFAULT_X1PLUS_SETTINGS = { }
+    DEFAULT_X1PLUS_SETTINGS = {
+        "boot.quick_boot": False,
+        "boot.dump_emmc": False,
+        "boot.sdcard_syslog": False,
+        "boot.perf_log": False,
+        "ota.enable": False,
+    }
 
     def __init__(self, router):
         self.router = router
 
         if IS_EMULATING:
+            self.settings_dir = "/"
             self.filename = "/tmp/x1plus-settings.json"
         else:
-            settings_dir = f"/mnt/sdcard/x1plus/printers/{_get_sn()}"
-            self.filename = f"{settings_dir}/settings.json"
-            os.makedirs(settings_dir, exist_ok = True)
-        
+            self.settings_dir = f"/mnt/sdcard/x1plus/printers/{_get_sn()}"
+            self.filename = f"{self.settings_dir}/settings.json"
+            os.makedirs(self.settings_dir, exist_ok=True)
+
         # Before we startup, do we have our settings file? Try to read, create if it doesn't exist.
         try:
-            with open(self.filename, 'r') as fh:
+            with open(self.filename, "r") as fh:
                 self.settings = json.load(fh)
         except FileNotFoundError as exc:
-            logger.warning('settings file does not exist; creating with defaults...')
-            # TODO: Add logic here (call helper function) to check for flag files, 
-            # and adjust our defaults to match
-            self.settings = copy.deepcopy(SettingsService.DEFAULT_X1PLUS_SETTINGS)
+            logger.warning("settings file does not exist; creating with defaults...")
+            self.settings = self._migrate_old_settings()
             self._save()
-    
+
+    def _migrate_old_settings(self):
+        """
+        Used to migrate init.d flag files to our new json on first run.
+        All bbl_screen settings MUST BE MIGRATED BY bbl_screen!
+        """
+        defaults = copy.deepcopy(SettingsService.DEFAULT_X1PLUS_SETTINGS)
+
+        # quick boot
+        if os.path.exists(f"{self.settings_dir}/quick-boot"):
+            defaults.update({"boot.quick_boot": True})
+            Path(f"{self.settings_dir}/quick-boot").unlink(missing_ok=True)
+        # dump emmc
+        if os.path.exists(f"{self.settings_dir}/dump-emmc"):
+            defaults.update({"boot.dump_emmc": True})
+            Path(f"{self.settings_dir}/dump-emmc").unlink(missing_ok=True)
+        # syslog to sd
+        if os.path.exists(f"{self.settings_dir}/logsd"):
+            defaults.update({"boot.sdcard_syslog": True})
+            Path(f"{self.settings_dir}/logsd").unlink(missing_ok=True)
+        # performance logging (debugging)
+        if os.path.exists(f"{self.settings_dir}/perf_log"):
+            defaults.update({"boot.perf_log": True})
+            Path(f"{self.settings_dir}/perf_log").unlink(missing_ok=True)
```

Comment:
DRY... how about something like...
```suggestion
        for fname,flag in [("quick-boot","boot.quick_boot"), ("dump-emmc","boot.dump_emmc"), ("logsd", "boot.sdcard_syslog"), ("perf_log", "boot.perf_log")}:
          if os.path.exists(f"{self.settings_dir}/{fname}")):
              defaults[flag] = True
              Path(f"{self.settings_dir}/{fname}").unlink(missing_ok = True)
```